kwok/cloudprovider/cloudprovider.go

```diff
@@ -0,0 +1,277 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kwok
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"math"
+	"math/rand"
+	"strings"
+
+	"github.com/docker/docker/pkg/namesgenerator"
+	"github.com/samber/lo"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"sigs.k8s.io/karpenter/pkg/apis/v1beta1"
+	"sigs.k8s.io/karpenter/pkg/cloudprovider"
+	"sigs.k8s.io/karpenter/pkg/scheduling"
+)
+
+const (
+	kwokLabelKey          = "kwok.x-k8s.io/node"
+	kwokLabelValue        = "fake"
+	nodeViewerLabelKey    = "eks-node-viewer/instance-price"
+	kwokPartitionLabelKey = "kwok-partition"
+)
+
+func NewCloudProvider(ctx context.Context, client client.Client, instanceTypes []*cloudprovider.InstanceType) *CloudProvider {
```

Comment:
```suggestion
func NewCloudProvider(ctx context.Context, kubeClient client.Client, instanceTypes []*cloudprovider.InstanceType) *CloudProvider {
```

nit: a little weird that that the argument doesn't match the member field